backEnd/routes/auth.py

```diff
@@ -1,17 +1,187 @@
-from fastapi import APIRouter
+from fastapi import Depends, status, HTTPException, APIRouter, Response
+from fastapi.security import OAuth2PasswordBearer, OAuth2PasswordRequestForm
+from datetime import datetime, timedelta
+from pydantic import BaseModel
+from jose import JWTError, jwt
+import logging
 
+# Setup JWT configurations
+SECRET_KEY = "secret_key"  # Use a secure and randomly generated secret key
+ALGORITHM = "HS256"
+ACCESS_TOKEN_EXPIRE_MINUTES = 30
+
+# OAuth2 Password flow for login
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="api/v1/auth/login")
+
+# User model for sign-up and login
+
+
+class User(BaseModel):
+    username: str
+    email: str
+    password: str
+
+# Database call to retrieve user data
+
+
+def get_user_from_db(username: str):
+    # TODO: Fetch user from database by username
+    # This function should return user details such as username and hashed password from the database
+    pass
+
+
+def save_user_to_db(user: User):
+    # TODO: Save the new user (username, email, hashed password) to the database
+    pass
+
+# JWT utility functions
+
+
+# JWT utility functions
+def create_access_token(data: dict, expires_delta: timedelta = None):
```

Comment:
Let's make expiration token optional, user can check 'keep logged in' checkbox so no need for the expiration.